sdk/identity/azure-identity/azure/identity/_credentials/silent.py

```diff
@@ -57,47 +60,69 @@ def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:
+    def _initialize_cache(self, is_cae: bool = False) -> Optional[TokenCache]:
+
+        # If no cache options were provided, the default cache will be used. This credential accepts the
+        # user's default cache regardless of whether it's encrypted. It doesn't create a new cache. If the
+        # default cache exists, the user must have created it earlier. If it's unencrypted, the user must
+        # have allowed that.
+        cache_options = self._cache_persistence_options or TokenCachePersistenceOptions(allow_unencrypted_storage=True)
+
+        if platform.system() not in {"Darwin", "Linux", "Windows"}:
+            raise CredentialUnavailableError(message="Shared token cache is not supported on this platform.")
+
+        if not self._cache and not is_cae:
             try:
-                # If no cache options were provided, the default cache will be used. This credential accepts the
-                # user's default cache regardless of whether it's encrypted. It doesn't create a new cache. If the
-                # default cache exists, the user must have created it earlier. If it's unencrypted, the user must
-                # have allowed that.
-                options = self._cache_persistence_options or TokenCachePersistenceOptions(
-                    allow_unencrypted_storage=True
-                )
-                self._cache = _load_persistent_cache(options)
+                self._cache = _load_persistent_cache(cache_options, cache_suffix=CACHE_NON_CAE_SUFFIX)
             except Exception:  # pylint:disable=broad-except
-                pass
+                return None
 
-        self._initialized = True
+        if not self._cae_cache and is_cae:
+            try:
+                self._cae_cache = _load_persistent_cache(cache_options, cache_suffix=CACHE_CAE_SUFFIX)
+            except Exception:  # pylint:disable=broad-except
+                return None
+
+        return self._cae_cache if is_cae else self._cache
 
     def _get_client_application(self, **kwargs: Any):
         tenant_id = resolve_tenant(
             self._tenant_id, additionally_allowed_tenants=self._additionally_allowed_tenants, **kwargs
         )
-        if tenant_id not in self._client_applications:
+
+        client_applications_map = self._client_applications
+        capabilities = None
+        token_cache = self._cache
+
+        if kwargs.get("enable_cae"):
+            client_applications_map = self._cae_client_applications
             # CP1 = can handle claims challenges (CAE)
-            capabilities = None if EnvironmentVariables.AZURE_IDENTITY_DISABLE_CP1 in os.environ else ["CP1"]
-            self._client_applications[tenant_id] = PublicClientApplication(
+            capabilities = ["CP1"]
+            token_cache = self._cae_cache
+
+        if tenant_id not in client_applications_map:
+            client_applications_map[tenant_id] = PublicClientApplication(
```

Comment:
Is it possible that there are two clients with name tenant_id but one is CAEed the other is not?

Reply:
Certainly possible. In that case, the tenant_id will exist in both the `self._client_applications` and `self._cae_client_applications` dictionary where one ClientApplication will be stored in each.